xapian-core/weight/lmweight.cc

```diff
@@ -176,8 +176,11 @@ LMWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount len,
 	 * case query term is present in the document.
 	 */
 	double weight_document = wdf_double / len_double;
-	weight_sum = 1 + ((((1 - param_smoothing1) / param_smoothing1) * 
-                (weight_document / weight_collection)));
+        if (param_smoothing1 == 0) 
+            weight_sum = weight_document;
+        else 
+            weight_sum = 1 + ((1 - param_smoothing1) / param_smoothing1) * 
+            weight_document / weight_collection;
```

Comment:
This is still misaligned - your current indentation suggests to the reader that `weight_document / weight_collection;` is a separate statement, which is misleading.

The rule for wrapping at a dyadic operator (`*` in this case) is to align the start of its two operands vertically.

(That's the Xapian rule at least, though I struggle to see another sensible rule for this case).